Add ability to comments on pages

mykytabatrak opened the pull request:
[Trello card](https://trello.com/c/lGiqfwE9)

mykytabatrak committed:
feat: added comment seeds

mykytabatrak committed:
feat: fetch all comments

mykytabatrak committed:
merge: develop to new/page-comments

mykytabatrak committed:
feat: add comment

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
feat: impoved page and user-permission seeds

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
fix: backend now returns comment with data about its author

mykytabatrak committed:
feat: added seeds for page content

mykytabatrak committed:
feat: added comment section to pages

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
fix: order comments from newest to oldest

mykytabatrak committed:
feat: comment and response addition implemented

mykytabatrak committed:
feat: added socket.io to server and client

mykytabatrak committed:
feat: implemnted comment update with socket.io

mykytabatrak committed:
chore: cleanup

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
refactor

ursarik commented on backend/package.json:
```diff
@@ -57,6 +57,7 @@
     "pino": "7.0.0-rc.1",
     "pino-pretty": "5.1.2",
     "reflect-metadata": "0.1.13",
+    "socket.io": "^4.1.3",
```
```suggestion
    "socket.io": "4.1.3",
```

ursarik commented on backend/src/data/seed-data/comment.data.ts:
```diff
@@ -1,4 +1,144 @@
 export const comments = [
+  {
+    id: '23a94b8b-850e-4dc1-8eb9-788f8cd152ac',
+    text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, aut.',
+    authorId: '9a25ad37-a647-4c1a-897f-59dbf275765d',
+    pageId: 'f8cd8ff3-d1de-47a5-820e-73a58368b22e',
+    parentCommentId: null,
```
```suggestion
    text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, aut.',
    authorId: '9a25ad37-a647-4c1a-897f-59dbf275765d',
    pageId: 'f8cd8ff3-d1de-47a5-820e-73a58368b22e',
    parentCommentId: null,
```

id is added by db by default

ursarik commented on backend/src/socket/handlers.ts:
```diff
@@ -0,0 +1,10 @@
+export const handlers = (socket: Socket): void => {
+  logger.info(`Socket connected ${socket.id}`);
+
+  socket.on('page/join', (pageId: string) => {
```
same about event names

ursarik commented on backend/src/services/comment.service.ts:
```diff
@@ -0,0 +1,46 @@
+
+  const comment = await commentRepository.findOneById(id);
+
+  io.to(pageId).emit('page/newComment', comment);
```
```suggestion
  io.to(pageId).emit(EVENT_NAMES.PAGE_NEW_COMMENT), comment);
```
what about moving event names to enum `EVENT_NAMES.ts` in `shared`? so in future we could import them in frontend and use instead of hardcoding them

ursarik commented on backend/.env.example:
```diff
@@ -2,6 +2,7 @@
 # APPLICATION
 #
 PORT=3001
+SOCKET_PORT=3002
```
why ws should be on a separate port?

aleksandr-kuharenko replied:
I cannot see answer or changes, but conversation marked as resolved
@SirRedrick

mykytabatrak replied:
Sorry, I spoke to @AlikRakhmonov at the begging of our meeting, he said it will be okay both ways (on separate port and on the same), so I decided to do it how it was in the mini-project, on a separate port

ursarik commented on frontend/package.json:
```diff
@@ -31,6 +31,7 @@
     "react-scripts": "4.0.3",
     "react-select": "4.3.1",
     "react-toastify": "7.0.4",
+    "socket.io-client": "^4.1.3",
```
```suggestion
    "socket.io-client": "4.1.3",
```

mykytabatrak committed:
chore: locket socket.io client and server versions

mykytabatrak committed:
refactor: refactored comment components and socket handlers

moved comment formed into separate component
made comment avatar size same as avatar size in header
moved socket events into a enum

mykytabatrak committed:
merge: develop into new/page-comments

mykytabatrak committed:
fix: fixes to styling

aleksandr-kuharenko reviewed (changes requested):
https://github.com/BinaryStudioAcademy/bsa-2021-infostack/pull/93#discussion_r686790626

mykytabatrak committed:
merge: develop into new/page-comments
